h2o-core/src/main/java/water/Key.java

```diff
@@ -467,12 +472,13 @@ public String valueClassSimple() {
   }
 
   @Override public int hashCode() { return _hash; }
+
+  public boolean equals(Key k) {
+    return k != null && _hash == k._hash && Arrays.equals(k._kb,_kb);
+  }
+
   @Override public boolean equals( Object o ) {
-    if( this == o ) return true;
-    if( o == null ) return false;
-    Key k = (Key)o;
-    if( _hash != k._hash ) return false;
-    return Arrays.equals(k._kb,_kb);
+    return this == o || (o != null && (o instanceof Key) && equals((Key)o));
```

Comment:
I don't quite understand this. If `o` is a `Key` then the compiler should have called the `public boolean equals(Key k)` method above. Is there a case when this method returns true then?